ui/src/Components/search/SearchController.vue

```diff
@@ -113,24 +113,11 @@
         number
         path
       }
+            }`)
 
-
-
-            }`).then( res => {
-        let i;
-        for (i = 0; i < res['data'].findBookByUniquePath.chapters.length; i++) {
-          const ch = new Chapter();
-          ch.hebrewNumeral = res['data'].findBookByUniquePath.chapters[i].hebrewNumeral;
-          ch.id = res['data'].findBookByUniquePath.chapters[i].id;
-          ch.number = res['data'].findBookByUniquePath.chapters[i].number;
-          ch.path = res['data'].findBookByUniquePath.chapters[i].path;
-          this.listOfChaptersInSelectedBook.push(ch);
-          //console.log(this.chaptersList);
-        }
```

Comment:
This whole section is waaaay overcomplicating this, also never define `i` outside of the actual for loop, you can do 
```
for (let i = 0, ....){...+
```
You do not need to instantiate anything just assert the type as I did. 
most modern JS will use `async/await` and not `promise.then` it much more readable.
use destrcturing it is your friend!

Reply:
As far as JS is concerned, perhaps not, but I think that TS requires it..I kept getting bugs until I did that...

Reply:
so async /await is better? I'll that

Reply:
Did you try this code? I think it will work without errors.
Either way:
- async/await is the way you should do, that is modern JS
- Ther very worst case should look like this:
```
data?.findBookByUniquePath?.chapters.forEach((chapter: Chapter) => {
          const newChapter = new Chapter(chapter);
          this.listOfChaptersInSelectedBook.push(newChapter)
      });
  ```
  But I am fairly certain that this is wrong and not necessary. 
  - NOTE: `new` keyword in JS is used primarily for instantiating singleton things not ever variables.